feat(spanv2): Double write to legacy attributes for backwards compatibility

Ahmed-Labs opened the pull request:
Double writes SpanV2 attributes to legacy attributes to achieve backwards compatibility. With both conventions present in a span, V2 spans will continue to be compatible with our products while we simultaneously start accumulating spans that conform to sentry conventions.

DB and HTTP spans both fall back to legacy attributes, this is done so that when the transactions span pipeline gets unified into the span first pipeline, they continue to be normalized.

See DACI for more information: https://www.notion.so/sentry/DACI-Backwards-compatibility-for-span-first-attribute-conventions-2c48b10e4b5d802dbfc3d90610b2537b?source=copy_link

Note: bumps sentry conventions

Ahmed-Labs committed:
double write to legacy attributes for backwards compatibility

Ahmed-Labs committed:
update changelog

Ahmed-Labs committed:
add test for db attribute normalization from legacy description attribute

Ahmed-Labs committed:
update changelog message

Ahmed-Labs committed:
add legacy http attributes to normalization test

Ahmed-Labs committed:
read from legacy attributes as a fall back during normalization

Ahmed-Labs committed:
fix typo

Ahmed-Labs committed:
move changelog message to unreleased

jjbayer commented:
@loewenheim could this be covered by generic double-write rules in sentry conventions instead?

Ahmed-Labs commented on relay-event-normalization/src/eap/mod.rs:
```diff
@@ -425,22 +425,32 @@ fn normalize_http_attributes(
     // This is equivalent to conditionally scrubbing by span category in the V1 pipeline.
-    if !attributes.contains_key(HTTP_REQUEST_METHOD) {
+    if !attributes.contains_key(HTTP_REQUEST_METHOD)
+        && !attributes.contains_key("http.request_method")
```
I just thought that since legacy attributes like `http.request_method` don't align with OTel and we're going in a direction where the SDKs won't be sending it at all, it would be counter productive to include it. I do also see value in adding it to conventions tho because that attribute will always be present in some spans (even with span first, there will be people with unupgraded SDKs I guess).

Ahmed-Labs committed:
bump sentry conventions

Ahmed-Labs committed:
replace hard coded attribute names with sentry conventions references

Ahmed-Labs commented:
Following up on this, I've added the attributes to sentry conventions ([PR](https://github.com/getsentry/sentry-conventions/pull/212)) and updated my changes accordingly.

**Note:** I wasn't able to add `http.request_method` because upon adding it, the conventions code gen created conflicting variable names with the existing `http.request.method` attribute (in the generated libraries) because it doesn't differentiate between periods and underscores in all cases. So it seems like the sentry conventions code gen needs to be updated, and I've created a [ticket](https://github.com/getsentry/sentry-conventions/issues/213) for it.

Ahmed-Labs committed:
move legacy http request method const to relay conventions
